tools/similarity_search/test/index.spec.ts

```diff
@@ -1,22 +1,98 @@
-import { SELF } from "cloudflare:test"
+import { SELF, env } from "cloudflare:test"
 import { describe, it, expect } from "vitest"
 
 import "../src/index"
 
+interface SimilarityCheckResponse {
+  similarity_score: number
+}
+
 describe("Authentication", () => {
-  it("returns 401 Unauthorized when API key is missing or invalid", async () => {
+  it("returns 401 Unauthorized when API key is invalid", async () => {
     const response = await SELF.fetch("https://example.com/", {
       method: "POST",
       headers: {
-        "Content-Type": "application/json"
+        "Content-Type": "application/json",
+        "X-API-Key": "invalid-api-key",
       },
       body: JSON.stringify({
         text: "Sample text",
-        namespace: "test-namespace"
-      })
+        namespace: "test-namespace",
+      }),
     })
 
     expect(response.status).toBe(401)
     expect(await response.text()).toBe("Unauthorized")
   })
 })
+
+describe("Validation", () => {
+  it("returns 400 Invalid JSON format when text is missing", async () => {
+    const response = await SELF.fetch("https://example.com/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-API-Key": "test-api-key",
+      },
+      body: JSON.stringify({
+        namespace: "test-namespace",
+      }),
+    })
+
+    expect(response.status).toBe(400)
+    expect(await response.text()).toBe("Invalid JSON format")
+  })
+
+  it("returns 400 Invalid JSON format when namespace is missing", async () => {
+    const response = await SELF.fetch("https://example.com/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-API-Key": "test-api-key",
+      },
+      body: JSON.stringify({
+        text: "Sample text",
+      }),
+    })
+
+    expect(response.status).toBe(400)
+    expect(await response.text()).toBe("Invalid JSON format")
+  })
+})
```

Comment:
can be merged together with [`.each`](https://vitest.dev/api/#test-each)

Reply:
Looks much better this way, fixed!